site/src/pages/TemplatePage/TemplateInsightsPage/TemplateInsightsPage.tsx

```diff
@@ -261,6 +281,209 @@ const TemplateUsagePanel = ({
   )
 }
 
+const TemplateParametersUsagePanel = ({
+  data,
+  ...panelProps
+}: PanelProps & {
+  data: TemplateInsightsResponse["report"]["parameters_usage"] | undefined
+}) => {
+  return (
+    <Panel {...panelProps}>
+      <PanelHeader>
+        <PanelTitle>Parameters usage</PanelTitle>
+        <PanelSubtitle>Last 7 days</PanelSubtitle>
+      </PanelHeader>
+      <PanelContent>
+        {!data && <Loader sx={{ height: 200 }} />}
+        {data && data.length === 0 && <NoDataAvailable sx={{ height: 200 }} />}
+        {data &&
+          data.length > 0 &&
+          data.map((parameter, parameterIndex) => {
```

Comment:
Can we wrap it into a dedicated React component, for instance, `ParameterUsageRow`?

Reply:
I usually extract code into a component when:
- It has a state
- It has too many conditionals
- It is large and not easy to read

So in this case, I would not do that to avoid nested components hell as we have for the Workspace component.

Reply:
> It is large and not easy to read

Yeah, I admit that this is my concern in terms of this PR, but maybe it isn't too hard. I saw a lot of conditional logic and assumed that it could be healthier to hide it behind a component. So, maybe not a state, but a lot of conditional logic.

Reply:
> but a lot of conditional logic.

I'm only seeing one for the label 🤔